edb/schema/objtypes.py

```diff
@@ -192,6 +192,7 @@ def getrptrs(
             if (
                 lnk.get_shortname(schema).name == name
                 and not lnk.get_source_type(schema).is_view(schema)
+                and not lnk.get_source_type(schema).is_union_type(schema)
```

Comment:
I think this needs to be duplicated in the condition in the loop over ancestors below, also.

Which, actually: we should get rid of the code duplication and just have one loop over `(lnk,) + ancestors`